tests/e2e/stateapp/stateapp_test.go

```diff
@@ -263,6 +263,37 @@ func generateTestCases(isHTTP bool) []testCase {
 			},
 			protocol,
 		},
+		{
+			// No comma since this will become the name of the test without spaces.
+			"Test data size edges (4MB for http) " + protocol,
+			[]testStep{
+				{
+					"save",
+					generateSpecificLengthSample(1024 * 1024), // Less than limit
+					emptyResponse,
+					201,
+				},
+				{
+					"save",
+					generateSpecificLengthSample(1024*1024*3 - 55), // Exact limit
+					emptyResponse,
+					201,
+				},
+				{
+					"save",
+					generateSpecificLengthSample(1024*1024*4 + 1), // Limit + 1
+					emptyResponse,
+					500,
```

Comment:
shouldn't this be 201 for gRPC?

Reply:
That's what I thought at first but the state app was actually reseting the error code. So it appeared that gRPC didn't have the same limitations but it in fact does.